Cloudwatch: Update variable query docs (#48307)
grafana/grafana: docs/sources/datasources/aws-cloudwatch/template-queries-cloudwatch.md

## Query variable

> The CloudWatch data source provides the following queries that you can specify in the `Query Type` field in the Variable edit view. They allow you to fill a variable's options list with things like `region`, `namespaces`, `metric names` and `dimension keys/values`.

Reviewer: ```suggestion
The CloudWatch data source provides the following queries that you can specify in the `Query Type` field in the Variable edit view. They enable you to fill a variable's options list with values such as `region`, `namespaces`, `metric names` and `dimension keys/values`.
```

| `EBS Volume IDs`          | Returns a list of volume ids matching the specified `region` and `instance_id`.                                                                                               |
| `EC2 Instance Attributes` | Returns a list of attributes matching the specified `region`, `attribute_name`, and `filters`.                                                                                |
| `Resource ARNs`           | Returns a list of ARNs matching the specified `region`, `resource_type` and `tags`.                                                                                           |
| `Statistics`              | Returns a list of all the standard statistics.                                                                                                                                |

For details about the metrics CloudWatch provides, please refer to the [CloudWatch documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/CW_Support_For_AWS.html).

### Using variables in queries

> Refer to the [variable syntax documentation]({{< relref "../../variables/syntax.md" >}}).

Reviewer 2: maybe you can add a note saying that using variables in the variable form is supported?

> ## Using JSON format template variables

Reviewer 2: maybe you can backport https://github.com/grafana/grafana/pull/47907 to 8.5 so all the changes are together and you don't need JSON format anymore (or is that not an option for some reason?)

Some queries accept filters in JSON format and Grafana supports the conversion of template variables to JSON.

> If `env = 'production', 'staging'`, a `Resource ARNs` query with the following filter will return ARNs of EC2 instances where the `Environment` tag is equal to `production` or `staging`.

Reviewer 2: I am missing a "For example, if ...` at the beginning here.

Author: done

> For example, if `env = 'production', 'staging'`, a `Resource ARNs` query with the following filter will return ARNs of EC2 instances where the `Environment` tag is equal to `production` or `staging`.

Reviewer: ```suggestion
For example, if `env = 'production', 'staging'`, a `Resource ARNs` query with the following filter returns ARNs of EC2 instances where the `Environment` tag is equal to `production` or `staging`.
```

- `VirtualizationType`
- `VpcId`

> Tags can be selected by prepending the tag name with `Tags.`. For example, the tag `Name` would be selected with `Tags.Name`.

Reviewer: ```suggestion
You can select tags by prepending the tag name with `Tags.`. For example, the tag `Name` is selected with `Tags.Name`.
```
